internal/pkg/service/common/etcdop/session_test.go

```diff
@@ -2,66 +2,159 @@ package etcdop
 
 import (
 	"context"
-	"runtime"
 	"sync"
 	"testing"
 	"time"
 
 	"github.com/cenkalti/backoff/v4"
+	"github.com/shopify/toxiproxy"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	"go.etcd.io/etcd/client/v3/concurrency"
-	"go.etcd.io/etcd/tests/v3/integration"
 
 	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/etcdhelper"
 )
 
-func TestResistantSession(t *testing.T) {
+func TestSession_Retries(t *testing.T) {
 	t.Parallel()
-	if runtime.GOOS != "linux" {
-		t.Skipf(`etcd session is tested only on Linux`)
-	}
 
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-	defer cancel()
 	wg := &sync.WaitGroup{}
 
-	// Create etcd cluster for test
-	integration.BeforeTestExternal(t)
-	cluster := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1, UseBridge: true})
-	defer cluster.Terminate(t)
-	cluster.WaitLeader(t)
-	member := cluster.Members[0]
-	client := cluster.Client(0)
+	// Get credentials
+	etcdCfg := etcdhelper.TmpNamespace(t)
+
+	// Setup proxy to drop etcd connection
+	proxy := toxiproxy.NewProxy()
+	proxy.Name = "etcd-bridge"
+	proxy.Upstream = etcdCfg.Endpoint
+	require.NoError(t, proxy.Start())
+	defer proxy.Stop()
```

Comment:
The etcd `integration` package is not stable (its interace can be modified anytime), it is used internally for etcd testing.

So I replaced it with the `toxiproxy`, it is a switchable proxy between test client and real `etcd` database.
It is simpler and better match real world.